aiosmtpd/smtp.py

```diff
@@ -64,7 +65,10 @@ def __init__(self, handler,
         self.seen_greeting = ''
         self.extended_smtp = False
         self.command_size_limits.clear()
-        self.fqdn = socket.getfqdn()   # XXX this blocks, fix it?
+        if not hostname:
+            self.hostname = socket.getfqdn()   # XXX this blocks, fix it?
```

Comment:
The XXX comment can probably be removed.  We know that `getfqdn()` can block because of the underlying libc calls.  I think it's not our job to avoid that, and this branch is doing the next best thing by providing an API to avoid the blocking when the client knows (through some other, out-of-band means) the hostname.

So let's remove the XXX comment and make sure we document this API for those purposes.